### PR TITLE
Document etcd3 migration process

### PR DESCRIPTION
**Pull request by justinsb**
Describe the process for adopting etcd3, including for calico users,
and including steps to perform more gradual migrations.

### CODE REVIEW

**Comment by justinsb** [comment]
cc @mikesplain

We'll likely need to review this in conjunction with the calico & etcd PRs... working on tidying those up now!

**Comment by chrisz100** [comment]
Issues with the e2e engine again? ...

File: docs/etcd3-migration.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+### Delay adopting etcd-manager with kops 1.12
+
+To delay adopting etcd-manager with kops 1.12, specify the provider as type `legacy`:
+
+```bash
+kops set cluster cluster.spec.etcdClusters[*].provider=legacy
+```
+
+To remove, `kops edit` your cluster and delete the `provider: Legacy` lines from both etcdCluster blocks.
+
```
Comment (chrisz100) [inline docs/etcd3-migration.md]: Etcd3 not etcd 3

**Comment by justinsb** [comment]
Thanks for the review @chrisz100 ... should be all fixed up I think.  Hoping e2e is better now as well!

**Comment by mikesplain** [comment]
Looks good to me! Thanks @justinsb!

/retest
/lgtm
/approve

**Comment by justinsb** [comment]
/retest

Just checking on a no-op PR if tests are back up!

**Comment by justinsb** [comment]
/retest

hope springs eternal

**Comment by justinsb** [comment]
/retest

hope may not be a strategy

**Pull request merged**